packages/core/src/components/toast/_toast.scss

```diff
@@ -103,7 +103,7 @@ $toast-intent-colors: (
   box-shadow: $pt-toast-box-shadow;
   display: flex;
   margin: $toast-margin 0 0;
-  max-width: $toast-max-width;
+  max-width: min($toast-max-width, calc(100vw - $toast-margin));
```

Comment:
probably should handle considering `$toast-min-width` here as well 🤔

Reply:
from https://www.w3.org/TR/CSS21/visudet.html#min-max-widths

> The following algorithm describes how the two properties influence the [used value](https://www.w3.org/TR/CSS21/cascade.html#computed-value) of the ['width'](https://www.w3.org/TR/CSS21/visudet.html#propdef-width) property:
> The tentative used width is calculated (without ['min-width'](https://www.w3.org/TR/CSS21/visudet.html#propdef-min-width) and ['max-width'](https://www.w3.org/TR/CSS21/visudet.html#propdef-max-width)) following the rules under ["Calculating widths and margins"](https://www.w3.org/TR/CSS21/visudet.html#Computing_widths_and_margins) above.
If the tentative used width is greater than ['max-width'](https://www.w3.org/TR/CSS21/visudet.html#propdef-max-width), the rules [above](https://www.w3.org/TR/CSS21/visudet.html#Computing_widths_and_margins) are applied again, but this time using the computed value of ['max-width'](https://www.w3.org/TR/CSS21/visudet.html#propdef-max-width) as the computed value for ['width'](https://www.w3.org/TR/CSS21/visudet.html#propdef-width).
If the resulting width is smaller than ['min-width'](https://www.w3.org/TR/CSS21/visudet.html#propdef-min-width), the rules [above](https://www.w3.org/TR/CSS21/visudet.html#Computing_widths_and_margins) are applied again, but this time using the value of ['min-width'](https://www.w3.org/TR/CSS21/visudet.html#propdef-min-width) as the computed value for ['width'](https://www.w3.org/TR/CSS21/visudet.html#propdef-width).

so I guess I should do something similar to this for the `min-width` field as well (though we're _propobably_ safe with 300px...)